test/service.test.ts

```diff
@@ -86,93 +82,27 @@ test('Can serve by port', () => {
 
 });
 
-test('Cannot add a deployment if the deployment does not have any containers', () => {
-
-  const chart = Testing.chart();
-
-  const service = new kplus.Service(chart, 'service');
-  const deployment = new kplus.Deployment(chart, 'dep');
-
-  // THEN
-  expect(() => service.addDeployment(deployment))
-    .toThrow(/Cannot expose a deployment without containers/);
-
-});
-
 test('Synthesizes spec lazily', () => {
 
   const chart = Testing.chart();
 
   const service = new kplus.Service(chart, 'Service');
 
-  service.addSelector('key', 'value');
-  service.serve(9000);
+  service.select(kplus.Pods.select(chart, 'Pods', { labels: { key: 'value' } }));
+  service.bind(9000);
 
   const spec = Testing.synth(chart)[0].spec;
   expect(spec.selector).toEqual({ key: 'value' });
   expect(spec.ports).toEqual([{ port: 9000 }]);
 
 });
 
-test('Can associate a deployment with an existing service', () => {
-
-  const chart = Testing.chart();
-
-  const service = new kplus.Service(chart, 'service');
-  const deployment = new kplus.Deployment(chart, 'dep');
-  deployment.addContainer({ image: 'foo', port: 7777 });
-
-  service.addDeployment(deployment);
-
-  const expectedSelector = { 'cdk8s.io/metadata.addr': 'test-dep-c8cc9f8f' };
```

Comment:
Is there still a way to achieve this? It would be nice to have a test for that, even if it's not explicitly using a hand-crafted method (`addDeployment`)

Reply:
Yeah I added a test, you basically just pass the deployment object to the `selector` property when creating the service, or in `service.select`.